File: src/backend/AgenticCompany.Core/Entities/Plan.cs
using AgenticCompany.Core.Enums;

namespace AgenticCompany.Core.Entities;

public class Plan
{
    public Guid Id { get; set; }
    public Guid SpecId { get; set; }
    public string Content { get; set; } = string.Empty;
    public PlanType PlanType { get; set; }
    public PlanStatus Status { get; set; } = PlanStatus.Draft;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Spec Spec { get; set; } = null!;
    public ICollection<TaskItem> Tasks { get; set; } = [];
}

File: src/frontend/src/api/schemas/plan.ts
import { z } from 'zod';

export const PlanStatus = z.enum(['Draft', 'Active', 'Completed', 'Archived']);
export type PlanStatus = z.infer<typeof PlanStatus>;

export const PlanType = z.enum(['Strategic', 'Technical']);
export type PlanType = z.infer<typeof PlanType>;

export const PlanSchema = z.object({
  id: z.string().uuid(),
  specId: z.string().uuid(),
  content: z.string(),
  planType: PlanType,
  status: PlanStatus,
  createdAt: z.string().datetime(),
  updatedAt: z.string().datetime(),
});

export type Plan = z.infer<typeof PlanSchema>;
